data/pigui/game.lua

```diff
@@ -0,0 +1,539 @@
+	end
+
+end
+
+-- heading, pitch and roll
+local function displayReticulePitchHorizonCompass(center)
+	local heading, pitch, roll = Game.player:GetHeadingPitchRoll("planet")
+	local pitch_degrees = (pitch / ui.twoPi * 360)
+	local heading_degrees = (heading / ui.twoPi * 360)
+	local roll_degrees = (roll / ui.twoPi * 360);
+	local size = 12
+
+	if showNavigationalNumbers then
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius - size * 4, 3)
+		ui.addStyledText(uiPos, math.floor(pitch_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.right, ui.anchor.center, "Current pitch")
+
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius - size * 2, 0)
+		ui.addStyledText(uiPos, math.floor(heading_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.center, ui.anchor.top, "Current heading")
+
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius, 6)
+		ui.addStyledText(uiPos, math.floor(roll_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.center, ui.anchor.top, "Current roll")
+	end
+
+	displayReticulePitch(center, pitch_degrees)
+	displayReticuleHorizon(center, roll_degrees)
+	displayReticuleCompass(center, heading_degrees)
+end
+
+-- show the small indicator inside the reticule circle
+local function displayIndicator(center, onscreen, position, direction, icon, color, showIndicator)
+	local size = 32
+	local iconSize = 16
+	local dir = direction * reticuleCircleRadius * 0.90
+	local indicator = center + dir
+	if onscreen then
+		ui.addIcon(position, icon, color, size, ui.anchor.center, ui.anchor.center)
+	end
+	if showIndicator and (center - position):magnitude() > reticuleCircleRadius * 1.2 then
+		ui.addIcon(indicator, icon, color, iconSize, ui.anchor.center, ui.anchor.center)
+	end
+end
+
+local reticuleTarget = "frame"
+
+local function displayTargetData(target, center, radius, combatTarget, navTarget, colorLight, colorDark, frameColor, navTargetColor, combatTargetColor)
+	local player = Game.player
+	if target then
+		local velocity = player:GetVelocityRelTo(target)
+		local position = player:GetPositionRelTo(target)
+
+		local uiPos = ui.pointOnClock(center, radius, 2)
+		-- label of target
+		local tooltip = reticuleTarget == "combatTarget" and lui.HUD_CURRENT_COMBAT_TARGET or (reticuleTarget == "navTarget" and lui.HUD_CURRENT_NAV_TARGET or lui.HUD_CURRENT_FRAME)
+		ui.addStyledText(uiPos, target.label, colorDark, ui.fonts.pionillium.medium, ui.anchor.left, ui.anchor.baseline, tooltip)
+
+		-- current distance, relative speed
+		uiPos = ui.pointOnClock(center, radius, 2.5)
+		local distance, distance_unit = ui.Format.Distance(player:DistanceTo(target))
+		local approach_speed = position:dot(velocity) / position:magnitude()
+		local speed, speed_unit = ui.Format.Speed(approach_speed)
+
+		ui.addFancyText(uiPos,
```

Comment:
refactored this as you suggested, makes way more sense than what I had before ;) thank you!